src/main/webapp/app/entities/text-assessments/text-assessments.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from 'app/app.constants';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Result } from 'src/main/webapp/app/entities/result';
+import { Participation } from 'app/entities/participation';
+import { Feedback } from 'app/entities/feedback';
+
+type EntityResponseType = HttpResponse<Result>;
+
+@Injectable({
+    providedIn: 'root'
+})
+export class TextAssessmentsService {
+    private readonly resourceUrl = SERVER_API_URL + 'api/text-assessments';
+
+    constructor(private http: HttpClient) {}
+
+    public save(textAssessments: Feedback[], exerciseId: number, resultId: number): Observable<EntityResponseType> {
+        return this.http
+            .put<Result>(`${this.resourceUrl}/exercise/${exerciseId}/result/${resultId}`, textAssessments, { observe: 'response' })
+            .map((res: EntityResponseType) => this.convertResponse(res));
+    }
+
+    public submit(textAssessments: Feedback[], exerciseId: number, resultId: number): Observable<EntityResponseType> {
+        return this.http
+            .put<Result>(`${this.resourceUrl}/exercise/${exerciseId}/result/${resultId}/submit`, textAssessments, { observe: 'response' })
+            .map((res: EntityResponseType) => this.convertResponse(res));
+    }
+
+    public getFeedbackDataForExerciseSubmission(exerciseId: number, submissionId: number): Observable<Participation> {
+        return this.http.get<Participation>(`${this.resourceUrl}/exercise/${exerciseId}/submission/${submissionId}`);
+    }
+
+    private convertResponse(res: EntityResponseType): EntityResponseType {
+        const body: Result = this.convertItemFromServer(res.body);
```

Comment:
please have a look at other Services, we need to transform date objects from the server into moment objects on the client. Result has the completionDate and in case it has a participation we need to transform the initializationDate as well

Reply:
see exercise.service.ts e.g.

Reply:
👍